packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1742,9 +1742,11 @@ class _TabBarState extends State<TabBar> {
         wrappedTabs[previousIndex] = _buildStyledTab(wrappedTabs[previousIndex], false, animation, _defaults);
       } else {
         // The user is dragging the TabBarView's PageView left or right.
-        final int tabIndex = _currentIndex!;
-        final Animation<double> centerAnimation = _DragAnimation(_controller!, tabIndex);
-        wrappedTabs[tabIndex] = _buildStyledTab(wrappedTabs[tabIndex], true, centerAnimation, _defaults);
+        if (_currentIndex! < widget.tabs.length) {
+          final int tabIndex = _currentIndex!;
```

Comment:
Might as well use that variable in the if-statement!

```suggestion
        final int tabIndex = _currentIndex!;
        if (tabIndex < widget.tabs.length) {
```

Reply:
> Might as well use that variable in the if-statement!

Variable `tabIndex` will also be used later and assigned to different values. Considering variables isolate and readable of codes I have put it all in one if statement.

Reply:
Thanks for the explanation.

These diffs on GitHub only show a few lines before & after the change, so I didn't see that this aligns with a pattern that the if-statements below are following.